cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -293,34 +295,52 @@ func scanOVAsOnNFS() (envelopes []Envelope, ovaPaths []string) {
 }
 
 func findOVAFiles(directory string) ([]string, []string, error) {
-	childs, err := os.ReadDir(directory)
-	if err != nil {
-		return nil, nil, err
-	}
-
 	var ovaFiles, ovfFiles []string
-	for _, child := range childs {
-		if !child.IsDir() {
-			continue
-		}
-		newDir := directory + "/" + child.Name()
-		files, err := os.ReadDir(newDir)
+	var ovaMaxDepth = 2
+
+	err := filepath.WalkDir(directory, func(path string, info os.DirEntry, err error) error {
 		if err != nil {
-			return nil, nil, err
+			return err
 		}
-		for _, file := range files {
-			path := filepath.Join(directory, child.Name(), file.Name())
-			switch {
-			case strings.HasSuffix(strings.ToLower(file.Name()), ".ova"):
-				ovaFiles = append(ovaFiles, path)
-			case strings.HasSuffix(strings.ToLower(file.Name()), ".ovf"):
-				ovfFiles = append(ovfFiles, path)
-			}
+
+		if info.IsDir() {
+			return nil
 		}
+
+		relativePath, _ := filepath.Rel(directory, path)
+		depth := len(strings.Split(relativePath, string(filepath.Separator)))
+
+		switch {
+		case (depth <= ovaMaxDepth) && isOva(info.Name()):
+			ovaFiles = append(ovaFiles, path)
+
+		case (depth >= ovaMaxDepth && depth <= ovaMaxDepth+1) && isOvf(info.Name()):
```

Comment:
there's a future scenario in which users upload an ova and then import it. the ova will usually be a tar file in this case, but for completeness - we can also inspect ovf files in the root folder

Reply:
yeah, it's different than what we talked about in the past - but with the current mechanism, it seems we can achieve this relatively easily and that would be even better than the solution we did in oVirt

Reply:
cool, changing to match